src/ImageSharp/Metadata/Profiles/IPTC/IptcProfile.cs

```diff
@@ -241,12 +252,25 @@ public void UpdateData()
                 length += value.Length + 5;
             }
 
+            bool hasValuesInUtf8 = this.HasValuesInUtf8();
+
+            if (hasValuesInUtf8)
+            {
+                // Additional length for UTF-8 Tag.
+                length += 5 + CodedCharacterSetUtf8Value.Length;
```

Comment:
The cool thing is that with the above suggestion the JIT will treat `CodedCharacterSetUtf8Value.Length` as constant (3) here.